File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using DevExpress.XtraBars.Ribbon;

namespace Navigator
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : RibbonForm
	{
		private System.Data.DataSet dataSet1;
		private System.Data.DataTable dataTable1;
		private System.Data.DataColumn dataColumn1;
        private System.Data.DataColumn dataColumn2;
		private DevExpress.XtraBars.BarButtonItem BarButtonFirst;
		private DevExpress.XtraBars.BarButtonItem BarButtonPrev;
        private DevExpress.XtraBars.BarButtonItem BarButtonNext;
        private DevExpress.XtraBars.BarButtonItem BarButtonLast;
        private DataGridView dataGridView1;
        private DataGridViewTextBoxColumn column1DataGridViewTextBoxColumn;
        private DataGridViewTextBoxColumn column2DataGridViewTextBoxColumn;
        private BindingSource bindingSource1;
        private DevExpress.XtraBars.Ribbon.RibbonControl ribbonControl1;
        private DevExpress.XtraBars.Ribbon.RibbonStatusBar ribbonStatusBar1;
        private DevExpress.XtraBars.BarButtonItem BarButtonAdd;
        private DevExpress.XtraBars.BarButtonItem BarButtonDelete;
        private DevExpress.XtraBars.BarButtonItem BarButtonEndEdit;
        private DevExpress.XtraBars.BarButtonItem BarButtonCancelEdit;
        private IContainer components;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.components = new System.ComponentModel.Container();
            this.dataSet1 = new System.Data.DataSet();
            this.dataTable1 = new System.Data.DataTable();
            this.dataColumn1 = new System.Data.DataColumn();
            this.dataColumn2 = new System.Data.DataColumn();
            this.BarButtonFirst = new DevExpress.XtraBars.BarButtonItem();
            this.BarButtonPrev = new DevExpress.XtraBars.BarButtonItem();
            this.BarButtonNext = new DevExpress.XtraBars.BarButtonItem();
            this.BarButtonLast = new DevExpress.XtraBars.BarButtonItem();
            this.dataGridView1 = new System.Windows.Forms.DataGridView();
            this.column1DataGridViewTextBoxColumn = new System.Windows.Forms.DataGridViewTextBoxColumn();
            this.column2DataGridViewTextBoxColumn = new System.Windows.Forms.DataGridViewTextBoxColumn();
            this.bindingSource1 = new System.Windows.Forms.BindingSource(this.components);
            this.ribbonControl1 = new DevExpress.XtraBars.Ribbon.RibbonControl();
            this.BarButtonAdd = new DevExpress.XtraBars.BarButtonItem();
            this.BarButtonDelete = new DevExpress.XtraBars.BarButtonItem();
            this.BarButtonEndEdit = new DevExpress.XtraBars.BarButtonItem();
            this.BarButtonCancelEdit = new DevExpress.XtraBars.BarButtonItem();
            this.ribbonStatusBar1 = new DevExpress.XtraBars.Ribbon.RibbonStatusBar();
            ((System.ComponentModel.ISupportInitialize)(this.dataSet1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.dataTable1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.dataGridView1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.bindingSource1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.ribbonControl1)).BeginInit();
            this.SuspendLayout();
            // 
            // dataSet1
            // 
            this.dataSet1.DataSetName = "NewDataSet";
            this.dataSet1.Locale = new System.Globalization.CultureInfo("en-US");
            this.dataSet1.Tables.AddRange(new System.Data.DataTable[] {
            this.dataTable1});
            // 
            // dataTable1
            // 
            this.dataTable1.Columns.AddRange(new System.Data.DataColumn[] {
            this.dataColumn1,
            this.dataColumn2});
            this.dataTable1.TableName = "Table1";
            // 
            // dataColumn1
            // 
            this.dataColumn1.ColumnName = "Column1";
            // 
            // dataColumn2
            // 
            this.dataColumn2.ColumnName = "Column2";
            // 
            // BarButtonFirst
            // 
            this.BarButtonFirst.Caption = "First";
            this.BarButtonFirst.Id = 0;
            this.BarButtonFirst.Name = "BarButtonFirst";
            // 
            // BarButtonPrev
            // 
            this.BarButtonPrev.Caption = "Prev";
            this.BarButtonPrev.Id = 1;
            this.BarButtonPrev.Name = "BarButtonPrev";
            // 
            // BarButtonNext
            // 
            this.BarButtonNext.Caption = "Next";
            this.BarButtonNext.Id = 2;
            this.BarButtonNext.Name = "BarButtonNext";
            // 
            // BarButtonLast
            // 
            this.BarButtonLast.Caption = "Last";
            this.BarButtonLast.Id = 3;
            this.BarButtonLast.Name = "BarButtonLast";
            // 
            // dataGridView1
            // 
            this.dataGridView1.AutoGenerateColumns = false;
            this.dataGridView1.ColumnHeadersHeightSizeMode = System.Windows.Forms.DataGridViewColumnHeadersHeightSizeMode.AutoSize;
            this.dataGridView1.Columns.AddRange(new System.Windows.Forms.DataGridViewColumn[] {
            this.column1DataGridViewTextBoxColumn,
            this.column2DataGridViewTextBoxColumn});
            this.dataGridView1.DataSource = this.bindingSource1;
            this.dataGridView1.Dock = System.Windows.Forms.DockStyle.Fill;
            this.dataGridView1.Location = new System.Drawing.Point(0, 50);
            this.dataGridView1.Name = "dataGridView1";
            this.dataGridView1.RowTemplate.Height = 24;
            this.dataGridView1.Size = new System.Drawing.Size(694, 275);
            this.dataGridView1.TabIndex = 4;
            // 
            // column1DataGridViewTextBoxColumn
            // 
            this.column1DataGridViewTextBoxColumn.DataPropertyName = "Column1";
            this.column1DataGridViewTextBoxColumn.HeaderText = "Column1";
            this.column1DataGridViewTextBoxColumn.Name = "column1DataGridViewTextBoxColumn";
            // 
            // column2DataGridViewTextBoxColumn
            // 
            this.column2DataGridViewTextBoxColumn.DataPropertyName = "Column2";
            this.column2DataGridViewTextBoxColumn.HeaderText = "Column2";
            this.column2DataGridViewTextBoxColumn.Name = "column2DataGridViewTextBoxColumn";
            // 
            // bindingSource1
            // 
            this.bindingSource1.DataMember = "Table1";
            this.bindingSource1.DataSource = this.dataSet1;
            // 
            // ribbonControl1
            // 
            this.ribbonControl1.ApplicationIcon = null;
            this.ribbonControl1.Items.AddRange(new DevExpress.XtraBars.BarItem[] {
            this.BarButtonFirst,
            this.BarButtonPrev,
            this.BarButtonNext,
            this.BarButtonLast,
            this.BarButtonAdd,
            this.BarButtonDelete,
            this.BarButtonEndEdit,
            this.BarButtonCancelEdit});
            this.ribbonControl1.Location = new System.Drawing.Point(0, 0);
            this.ribbonControl1.MaxItemId = 4;
            this.ribbonControl1.Name = "ribbonControl1";
            this.ribbonControl1.Size = new System.Drawing.Size(694, 50);
            this.ribbonControl1.StatusBar = this.ribbonStatusBar1;
            // 
            // BarButtonAdd
            // 
            this.BarButtonAdd.Caption = "Add";
            this.BarButtonAdd.Id = 0;
            this.BarButtonAdd.Name = "BarButtonAdd";
            // 
            // BarButtonDelete
            // 
            this.BarButtonDelete.Caption = "Delete";
            this.BarButtonDelete.Id = 1;
            this.BarButtonDelete.Name = "BarButtonDelete";
            // 
            // BarButtonEndEdit
            // 
            this.BarButtonEndEdit.Caption = "EndEdit";
            this.BarButtonEndEdit.Id = 2;
            this.BarButtonEndEdit.Name = "BarButtonEndEdit";
            // 
            // BarButtonCancelEdit
            // 
            this.BarButtonCancelEdit.Caption = "CancelEdit";
            this.BarButtonCancelEdit.Id = 3;
            this.BarButtonCancelEdit.Name = "BarButtonCancelEdit";
            // 
            // ribbonStatusBar1
            // 
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonFirst);
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonPrev);
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonNext);
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonLast);
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonAdd);
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonDelete);
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonEndEdit);
            this.ribbonStatusBar1.ItemLinks.Add(this.BarButtonCancelEdit);
            this.ribbonStatusBar1.Location = new System.Drawing.Point(0, 298);
            this.ribbonStatusBar1.Name = "ribbonStatusBar1";
            this.ribbonStatusBar1.Ribbon = this.ribbonControl1;
            this.ribbonStatusBar1.Size = new System.Drawing.Size(694, 27);
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(6, 16);
            this.ClientSize = new System.Drawing.Size(694, 325);
            this.Controls.Add(this.ribbonStatusBar1);
            this.Controls.Add(this.dataGridView1);
            this.Controls.Add(this.ribbonControl1);
            this.Name = "Form1";
            this.Ribbon = this.ribbonControl1;
            this.StatusBar = this.ribbonStatusBar1;
            this.Text = "Form1";
            this.Load += new System.EventHandler(this.Form1_Load);
            ((System.ComponentModel.ISupportInitialize)(this.dataSet1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.dataTable1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.dataGridView1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.bindingSource1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.ribbonControl1)).EndInit();
            this.ResumeLayout(false);

        }
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		private void CreateNavigator() {
			DevExpress.Demos.Navigator.XtraBarsDataNavigator navigator;
			navigator = new DevExpress.Demos.Navigator.XtraBarsDataNavigator();
			navigator.BindingContext = this.BindingContext;
			navigator.DataSource = bindingSource1;
            navigator.BarManager = ribbonControl1.Manager;

			// link buttons
			DevExpress.Demos.Navigator.ButtonLink link;
			link = navigator.Buttons.Add();
			link.ActionType = DevExpress.Demos.Navigator.ActionType.First;
			link.Button = BarButtonFirst;
			link = navigator.Buttons.Add();
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Prev;
			link.Button = BarButtonPrev;
			link = navigator.Buttons.Add();
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Next;
			link.Button = BarButtonNext;
			link = navigator.Buttons.Add();
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Last;
			link.Button = BarButtonLast;
            link = navigator.Buttons.Add();
            link.ActionType = DevExpress.Demos.Navigator.ActionType.Add;
            link.Button = BarButtonAdd;
            link = navigator.Buttons.Add();
            link.ActionType = DevExpress.Demos.Navigator.ActionType.Delete;
            link.Button = BarButtonDelete;
            link = navigator.Buttons.Add();
            link.ActionType = DevExpress.Demos.Navigator.ActionType.EndEdit;
            link.Button = BarButtonEndEdit;
            link = navigator.Buttons.Add();
            link.ActionType = DevExpress.Demos.Navigator.ActionType.CancelEdit;
            link.Button = BarButtonCancelEdit;
		}

		private void Form1_Load(object sender, System.EventArgs e) {
			
			for(int i = 0; i < 5; i++)
				dataTable1.Rows.Add(new object[] {"Item " + (i+1).ToString()});

            ribbonControl1.ForceInitialize();            
            CreateNavigator();			
		}
	}
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.XtraBars.Ribbon

Namespace Navigator
	''' <summary>
	''' Summary description for Form1.
	''' </summary>
	Public Class Form1
		Inherits RibbonForm
		Private dataSet1 As System.Data.DataSet
		Private dataTable1 As System.Data.DataTable
		Private dataColumn1 As System.Data.DataColumn
		Private dataColumn2 As System.Data.DataColumn
		Private BarButtonFirst As DevExpress.XtraBars.BarButtonItem
		Private BarButtonPrev As DevExpress.XtraBars.BarButtonItem
		Private BarButtonNext As DevExpress.XtraBars.BarButtonItem
		Private BarButtonLast As DevExpress.XtraBars.BarButtonItem
		Private dataGridView1 As DataGridView
		Private column1DataGridViewTextBoxColumn As DataGridViewTextBoxColumn
		Private column2DataGridViewTextBoxColumn As DataGridViewTextBoxColumn
		Private bindingSource1 As BindingSource
		Private ribbonControl1 As DevExpress.XtraBars.Ribbon.RibbonControl
		Private ribbonStatusBar1 As DevExpress.XtraBars.Ribbon.RibbonStatusBar
		Private BarButtonAdd As DevExpress.XtraBars.BarButtonItem
		Private BarButtonDelete As DevExpress.XtraBars.BarButtonItem
		Private BarButtonEndEdit As DevExpress.XtraBars.BarButtonItem
		Private BarButtonCancelEdit As DevExpress.XtraBars.BarButtonItem
		Private components As IContainer

		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()

			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub

		''' <summary>
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If components IsNot Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.components = New System.ComponentModel.Container()
			Me.dataSet1 = New System.Data.DataSet()
			Me.dataTable1 = New System.Data.DataTable()
			Me.dataColumn1 = New System.Data.DataColumn()
			Me.dataColumn2 = New System.Data.DataColumn()
			Me.BarButtonFirst = New DevExpress.XtraBars.BarButtonItem()
			Me.BarButtonPrev = New DevExpress.XtraBars.BarButtonItem()
			Me.BarButtonNext = New DevExpress.XtraBars.BarButtonItem()
			Me.BarButtonLast = New DevExpress.XtraBars.BarButtonItem()
			Me.dataGridView1 = New System.Windows.Forms.DataGridView()
			Me.column1DataGridViewTextBoxColumn = New System.Windows.Forms.DataGridViewTextBoxColumn()
			Me.column2DataGridViewTextBoxColumn = New System.Windows.Forms.DataGridViewTextBoxColumn()
			Me.bindingSource1 = New System.Windows.Forms.BindingSource(Me.components)
			Me.ribbonControl1 = New DevExpress.XtraBars.Ribbon.RibbonControl()
			Me.BarButtonAdd = New DevExpress.XtraBars.BarButtonItem()
			Me.BarButtonDelete = New DevExpress.XtraBars.BarButtonItem()
			Me.BarButtonEndEdit = New DevExpress.XtraBars.BarButtonItem()
			Me.BarButtonCancelEdit = New DevExpress.XtraBars.BarButtonItem()
			Me.ribbonStatusBar1 = New DevExpress.XtraBars.Ribbon.RibbonStatusBar()
			CType(Me.dataSet1, System.ComponentModel.ISupportInitialize).BeginInit()
			CType(Me.dataTable1, System.ComponentModel.ISupportInitialize).BeginInit()
			CType(Me.dataGridView1, System.ComponentModel.ISupportInitialize).BeginInit()
			CType(Me.bindingSource1, System.ComponentModel.ISupportInitialize).BeginInit()
			CType(Me.ribbonControl1, System.ComponentModel.ISupportInitialize).BeginInit()
			Me.SuspendLayout()
			' 
			' dataSet1
			' 
			Me.dataSet1.DataSetName = "NewDataSet"
			Me.dataSet1.Locale = New System.Globalization.CultureInfo("en-US")
			Me.dataSet1.Tables.AddRange(New System.Data.DataTable() { Me.dataTable1})
			' 
			' dataTable1
			' 
			Me.dataTable1.Columns.AddRange(New System.Data.DataColumn() { Me.dataColumn1, Me.dataColumn2})
			Me.dataTable1.TableName = "Table1"
			' 
			' dataColumn1
			' 
			Me.dataColumn1.ColumnName = "Column1"
			' 
			' dataColumn2
			' 
			Me.dataColumn2.ColumnName = "Column2"
			' 
			' BarButtonFirst
			' 
			Me.BarButtonFirst.Caption = "First"
			Me.BarButtonFirst.Id = 0
			Me.BarButtonFirst.Name = "BarButtonFirst"
			' 
			' BarButtonPrev
			' 
			Me.BarButtonPrev.Caption = "Prev"
			Me.BarButtonPrev.Id = 1
			Me.BarButtonPrev.Name = "BarButtonPrev"
			' 
			' BarButtonNext
			' 
			Me.BarButtonNext.Caption = "Next"
			Me.BarButtonNext.Id = 2
			Me.BarButtonNext.Name = "BarButtonNext"
			' 
			' BarButtonLast
			' 
			Me.BarButtonLast.Caption = "Last"
			Me.BarButtonLast.Id = 3
			Me.BarButtonLast.Name = "BarButtonLast"
			' 
			' dataGridView1
			' 
			Me.dataGridView1.AutoGenerateColumns = False
			Me.dataGridView1.ColumnHeadersHeightSizeMode = System.Windows.Forms.DataGridViewColumnHeadersHeightSizeMode.AutoSize
			Me.dataGridView1.Columns.AddRange(New System.Windows.Forms.DataGridViewColumn() { Me.column1DataGridViewTextBoxColumn, Me.column2DataGridViewTextBoxColumn})
			Me.dataGridView1.DataSource = Me.bindingSource1
			Me.dataGridView1.Dock = System.Windows.Forms.DockStyle.Fill
			Me.dataGridView1.Location = New System.Drawing.Point(0, 50)
			Me.dataGridView1.Name = "dataGridView1"
			Me.dataGridView1.RowTemplate.Height = 24
			Me.dataGridView1.Size = New System.Drawing.Size(694, 275)
			Me.dataGridView1.TabIndex = 4
			' 
			' column1DataGridViewTextBoxColumn
			' 
			Me.column1DataGridViewTextBoxColumn.DataPropertyName = "Column1"
			Me.column1DataGridViewTextBoxColumn.HeaderText = "Column1"
			Me.column1DataGridViewTextBoxColumn.Name = "column1DataGridViewTextBoxColumn"
			' 
			' column2DataGridViewTextBoxColumn
			' 
			Me.column2DataGridViewTextBoxColumn.DataPropertyName = "Column2"
			Me.column2DataGridViewTextBoxColumn.HeaderText = "Column2"
			Me.column2DataGridViewTextBoxColumn.Name = "column2DataGridViewTextBoxColumn"
			' 
			' bindingSource1
			' 
			Me.bindingSource1.DataMember = "Table1"
			Me.bindingSource1.DataSource = Me.dataSet1
			' 
			' ribbonControl1
			' 
			Me.ribbonControl1.ApplicationIcon = Nothing
			Me.ribbonControl1.Items.AddRange(New DevExpress.XtraBars.BarItem() { Me.BarButtonFirst, Me.BarButtonPrev, Me.BarButtonNext, Me.BarButtonLast, Me.BarButtonAdd, Me.BarButtonDelete, Me.BarButtonEndEdit, Me.BarButtonCancelEdit})
			Me.ribbonControl1.Location = New System.Drawing.Point(0, 0)
			Me.ribbonControl1.MaxItemId = 4
			Me.ribbonControl1.Name = "ribbonControl1"
			Me.ribbonControl1.Size = New System.Drawing.Size(694, 50)
			Me.ribbonControl1.StatusBar = Me.ribbonStatusBar1
			' 
			' BarButtonAdd
			' 
			Me.BarButtonAdd.Caption = "Add"
			Me.BarButtonAdd.Id = 0
			Me.BarButtonAdd.Name = "BarButtonAdd"
			' 
			' BarButtonDelete
			' 
			Me.BarButtonDelete.Caption = "Delete"
			Me.BarButtonDelete.Id = 1
			Me.BarButtonDelete.Name = "BarButtonDelete"
			' 
			' BarButtonEndEdit
			' 
			Me.BarButtonEndEdit.Caption = "EndEdit"
			Me.BarButtonEndEdit.Id = 2
			Me.BarButtonEndEdit.Name = "BarButtonEndEdit"
			' 
			' BarButtonCancelEdit
			' 
			Me.BarButtonCancelEdit.Caption = "CancelEdit"
			Me.BarButtonCancelEdit.Id = 3
			Me.BarButtonCancelEdit.Name = "BarButtonCancelEdit"
			' 
			' ribbonStatusBar1
			' 
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonFirst)
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonPrev)
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonNext)
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonLast)
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonAdd)
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonDelete)
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonEndEdit)
			Me.ribbonStatusBar1.ItemLinks.Add(Me.BarButtonCancelEdit)
			Me.ribbonStatusBar1.Location = New System.Drawing.Point(0, 298)
			Me.ribbonStatusBar1.Name = "ribbonStatusBar1"
			Me.ribbonStatusBar1.Ribbon = Me.ribbonControl1
			Me.ribbonStatusBar1.Size = New System.Drawing.Size(694, 27)
			' 
			' Form1
			' 
			Me.AutoScaleBaseSize = New System.Drawing.Size(6, 16)
			Me.ClientSize = New System.Drawing.Size(694, 325)
			Me.Controls.Add(Me.ribbonStatusBar1)
			Me.Controls.Add(Me.dataGridView1)
			Me.Controls.Add(Me.ribbonControl1)
			Me.Name = "Form1"
			Me.Ribbon = Me.ribbonControl1
			Me.StatusBar = Me.ribbonStatusBar1
			Me.Text = "Form1"
'			Me.Load += New System.EventHandler(Me.Form1_Load);
			CType(Me.dataSet1, System.ComponentModel.ISupportInitialize).EndInit()
			CType(Me.dataTable1, System.ComponentModel.ISupportInitialize).EndInit()
			CType(Me.dataGridView1, System.ComponentModel.ISupportInitialize).EndInit()
			CType(Me.bindingSource1, System.ComponentModel.ISupportInitialize).EndInit()
			CType(Me.ribbonControl1, System.ComponentModel.ISupportInitialize).EndInit()
			Me.ResumeLayout(False)

		End Sub
		#End Region

		''' <summary>
		''' The main entry point for the application.
		''' </summary>
		<STAThread> _
		Shared Sub Main()
			Application.Run(New Form1())
		End Sub

		Private Sub CreateNavigator()
			Dim navigator As DevExpress.Demos.Navigator.XtraBarsDataNavigator
			navigator = New DevExpress.Demos.Navigator.XtraBarsDataNavigator()
			navigator.BindingContext = Me.BindingContext
			navigator.DataSource = bindingSource1
			navigator.BarManager = ribbonControl1.Manager

			' link buttons
			Dim link As DevExpress.Demos.Navigator.ButtonLink
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.First
			link.Button = BarButtonFirst
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Prev
			link.Button = BarButtonPrev
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Next
			link.Button = BarButtonNext
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Last
			link.Button = BarButtonLast
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Add
			link.Button = BarButtonAdd
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.Delete
			link.Button = BarButtonDelete
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.EndEdit
			link.Button = BarButtonEndEdit
			link = navigator.Buttons.Add()
			link.ActionType = DevExpress.Demos.Navigator.ActionType.CancelEdit
			link.Button = BarButtonCancelEdit
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load

			For i As Integer = 0 To 4
				dataTable1.Rows.Add(New Object() {"Item " & (i+1).ToString()})
			Next i

			ribbonControl1.ForceInitialize()
			CreateNavigator()
		End Sub
	End Class
End Namespace
